src/main/java/picard/illumina/parser/readers/CbclReader.java

```diff
@@ -57,117 +63,189 @@
+                    for (int j = 0; j < numberOfBins; j++) {
+                        headerBuffer.getInt(); // first int is "from" value, which we don't need
+                        int to = headerBuffer.getInt();
+                        qualityBins[j] = (byte) to;
+                    }
+                    long filePos = 0;
+
+                    int numTiles = headerBuffer.getInt();
+                    TileData tileInfo = null;
+                    for (int j = 0; j < numTiles; j++) {
+                        int tile = headerBuffer.getInt();
+                        int numClustersInTile = headerBuffer.getInt();
+                        int uncompressedBlockSize = headerBuffer.getInt();
+                        int compressedBlockSize = headerBuffer.getInt();
+                        TileData tileData = new TileData(tile, numClustersInTile, uncompressedBlockSize, compressedBlockSize, filePos);
+                        allTiles.get(i + 1).add(tileData);
+                        if (tile == tileNum) {
+                            tileInfo = tileData;
+                        }
+                        filePos += compressedBlockSize;
+                    }
 
-                int numTiles = headerBuffer.getInt();
-                TileData[] tileInfo = new TileData[numTiles];
-                for (int j = 0; j < numTiles; j++) {
-                    int tileNum = headerBuffer.getInt();
-                    int numClustersInTile = headerBuffer.getInt();
-                    int uncompressedBlockSize = headerBuffer.getInt();
-                    int compressedBlockSize = headerBuffer.getInt();
-                    tileInfo[j] = new TileData(tileNum, numClustersInTile, uncompressedBlockSize, compressedBlockSize);
+                    boolean pfExcluded = headerBuffer.get() == 1;
+                    //try the next surface if we didn't find the tile
+                    if (tileInfo == null) continue;
+
+                    cycleData[i] = new CycleData(version, headerSize, bitsPerBasecall, bitsPerQualityScore, numberOfBins, qualityBins, numTiles, tileInfo, pfExcluded);
+                    this.streams[i] = stream;
+                    this.streamFiles[i] = bclFile;
+                    byteBuffer.clear();
+                    headerBuffer.clear();
                 }
+            }
+
+            if (headerOnly) return;
 
-                boolean pfExcluded = headerBuffer.get() == 1;
-                cycleData[i] = new CycleData(version, headerSize, bitsPerBasecall, bitsPerQualityScore, numberOfBins, qualityBins, numTiles, tileInfo, pfExcluded);
-                this.streams[i] = stream;
-                this.streamFiles[i] = bclFile;
-                byteBuffer.clear();
-                headerBuffer.clear();
+            int totalCycleCount = 0;
+
+            if (cycleData[totalCycleCount].tileInfo == null) {
+                throw new PicardException("Could not find tile " + cycleData[totalCycleCount].tileInfo);
```

Comment:
If I'm reading this if+exception message correctly, `"Could not find tile " + cycleData[totalCycleCount].tileInfo` will always translate into `"Could not find tile null"`. Is there an alternative set of variables for the exception message?